File: app/src/main/java/com/barryyang/barryyangdemo/java/reflection/Student.java
package com.barryyang.barryyangdemo.java.reflection;

/**
 * @author : BarryYang
 * @date : 2021/6/22 3:45 下午
 * @desc :
 */
public class Student {

    private int age;
    public String userName;

    public Student() {

    }

    private Student(String userName) {
        this.userName = userName;
    }

    private void setAge(int age) {
        this.age = age;
    }

    private String show(String message) {
        System.out.println("userName = " + userName + ",age = " + age + ",message = " + message);
        return "finish";
    }

}

File: app/src/main/java/com/barryyang/barryyangdemo/kotlin/lambda/Student.kt
package com.barryyang.barryyangdemo.kotlin.lambda

/**
 * @author : BarryYang
 * @date : 2020/8/19 3:39 PM
 * @desc :
 */
data class Student(val name: String, val age: Int, val sex: String, val score: Int) {

    val firstProperty = "First property: $name".also(::println)

    init {
        println("First initializer block that prints ${name}")
        println("First initializer block that prints ${firstProperty}")
    }

    val secondProperty = "Second property: ${name.length}".also(::println)

    init {
        println("Second initializer block that prints ${name.length}")
        println("First initializer block that prints ${secondProperty}")
    }
}